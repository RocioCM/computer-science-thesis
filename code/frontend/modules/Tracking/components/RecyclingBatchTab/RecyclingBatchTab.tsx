import React from 'react';
import { RecyclingBatch } from '../../types';

interface Props {
  batch?: RecyclingBatch | null;
}

const RecyclingBatchTab: React.FC<Props> = ({ batch }) => {
  return batch ? (
    <div className="h-full w-full overflow-auto hide-scroll relative">
      {/* Información General */}
      <div className="flex flex-col gap-s">
        <h3 className="text-lg font-semibold border-b pb-2">
          Información General
        </h3>
        <p>
          <span className="font-semibold">Material:</span> {batch.materialType}
        </p>
        <p>
          <span className="font-semibold">Tamaño:</span> {batch.size}
        </p>
        <p>
          <span className="font-semibold">Peso:</span> {batch.weight} kg
        </p>
        <p>
          <span className="font-semibold">Fecha de creación:</span>{' '}
          {new Date(batch.createdAt).toLocaleDateString()}
        </p>
      </div>

      {/* Composición */}
      <div className="mt-6">
        <h3 className="text-lg font-semibold border-t py-2">Composición</h3>
        {batch.composition.length > 0 ? (
          <table className="w-full mt-2 border border-gray-300 rounded-md">
            <thead>
              <tr className="bg-gray-100 text-gray-700">
                <th className="px-4 py-2 text-left">Material</th>
                <th className="px-4 py-2 text-left">Cantidad</th>
                <th className="px-4 py-2 text-left">Unidad</th>
              </tr>
            </thead>
            <tbody>
              {batch.composition.map((material, index) => (
                <tr key={index} className="border-t">
                  <td className="px-4 py-2">{material.name}</td>
                  <td className="px-4 py-2">{material.amount}</td>
                  <td className="px-4 py-2">{material.measureUnit}</td>
                </tr>
              ))}
            </tbody>
          </table>
        ) : (
          <p className="text-gray-600 mt-2">
            No hay información de composición.
          </p>
        )}
      </div>
      {/* Información Adicional */}
      {batch.extraInfo && (
        <div className="mt-6">
          <h3 className="text-lg font-semibold border-t py-2">
            Información Adicional
          </h3>
          <p className="text-gray-700 mt-2">{batch.extraInfo}</p>
        </div>
      )}
    </div>
  ) : null;
};

export default RecyclingBatchTab;
