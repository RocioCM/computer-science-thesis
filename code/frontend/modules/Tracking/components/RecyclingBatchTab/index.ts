import RecyclingBatchTab from './RecyclingBatchTab';

export default RecyclingBatchTab;
