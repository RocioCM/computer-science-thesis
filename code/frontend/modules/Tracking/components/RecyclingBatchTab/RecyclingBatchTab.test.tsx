import { render, screen, fireEvent, act } from '@testing-library/react';
import RecyclingBatchTab from './RecyclingBatchTab';

describe('Tracking - RecyclingBatchTab', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <RecyclingBatchTab
        batch={{
          id: 1,
          buyerOwner: '0x1234567890abcdef1234567890abcdef12345678',
          creator: '0xabcdefabcdefabcdefabcdefabcdefabcdefabcdef',
          extraInfo: 'Some extra info',
          materialType: 'Pure glass',
          wasteBottleIds: [1, 2, 3],
          size: '10x10x10',
          weight: 100,
          composition: [{ name: 'Glass', amount: 100, measureUnit: '%' }],
          createdAt: new Date().toISOString(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
