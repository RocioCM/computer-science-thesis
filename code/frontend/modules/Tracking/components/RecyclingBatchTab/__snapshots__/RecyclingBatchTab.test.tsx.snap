// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking - RecyclingBatchTab Matches snapshot 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative"
  >
    <div
      class="flex flex-col gap-s"
    >
      <h3
        class="text-lg font-semibold border-b pb-2"
      >
        Información General
      </h3>
      <p>
        <span
          class="font-semibold"
        >
          Material:
        </span>
         
        Pure glass
      </p>
      <p>
        <span
          class="font-semibold"
        >
          Tamaño:
        </span>
         
        10x10x10
      </p>
      <p>
        <span
          class="font-semibold"
        >
          Peso:
        </span>
         
        100
         kg
      </p>
      <p>
        <span
          class="font-semibold"
        >
          Fecha de creación:
        </span>
         
        13/4/2025
      </p>
    </div>
    <div
      class="mt-6"
    >
      <h3
        class="text-lg font-semibold border-t py-2"
      >
        Composición
      </h3>
      <table
        class="w-full mt-2 border border-gray-300 rounded-md"
      >
        <thead>
          <tr
            class="bg-gray-100 text-gray-700"
          >
            <th
              class="px-4 py-2 text-left"
            >
              Material
            </th>
            <th
              class="px-4 py-2 text-left"
            >
              Cantidad
            </th>
            <th
              class="px-4 py-2 text-left"
            >
              Unidad
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-t"
          >
            <td
              class="px-4 py-2"
            >
              Glass
            </td>
            <td
              class="px-4 py-2"
            >
              100
            </td>
            <td
              class="px-4 py-2"
            >
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="mt-6"
    >
      <h3
        class="text-lg font-semibold border-t py-2"
      >
        Información Adicional
      </h3>
      <p
        class="text-gray-700 mt-2"
      >
        Some extra info
      </p>
    </div>
  </div>
</div>
`;
